### PR TITLE
Coding Standards from Ticket 47632

### PR DESCRIPTION
**Pull request by mattyrob**
## Description
This PR includes all relevant changes included in upstream ticket 47632:
https://core.trac.wordpress.org/ticket/47632
Not all of the 27 commits were necessary or needed.

## Motivation and context
Improve coding standards for readability and to aid move to PHP 8 and 8.1 compatibility.
It also commences work as described in the discussion on #1098 

## How has this been tested?
Coding standards and PHPUnit tests run and pass locally after these changes.

## Screenshots
N/A

## Types of changes
- Bug fix / Enhancement

### CODE REVIEW

File: src/wp-includes/template-loader.php [hunk 1]
```diff
@@ -27,58 +27,79 @@
...
+	// Loop through each of the template conditionals, and find the appropriate template file.
+	foreach ( $tag_templates as $tag => $template_getter ) {
+		if ( call_user_func( $tag ) ) {
+			$template = call_user_func( $template_getter );
+		}
+
+		if ( $template ) {
+			if ( 'is_attachment' === $tag ) {
+				remove_filter( 'the_content', 'prepend_attachment' );
+			}
```
Comment (xxsimoxx) [inline src/wp-includes/template-loader.php:78]: `is_privacy_policy` and `get_privacy_policy_template` are since WP-5.2.
The others are OK.

**Review by xxsimoxx — CHANGES_REQUESTED** [review]
Another warning: `function 'is_privacy_policy' not found or invalid function name in src/wp-includes/template-loader.php on line 71`

**Review by xxsimoxx — APPROVED** [review]
Tested (together with #1117) on a live site with `WP_DEBUG` turned on, done common tasks and left in place for 3 days without any error or warning.

**Comment by mattyrob** [comment]
Discussed at weekly PR review and agreed for merge.

**Pull request merged**